### PR TITLE
added ref to map element

### PR DESCRIPTION
**Pull request by digvijaysrathore**
Added ref to the map element as per issue #32

### CODE REVIEW

File: src/components/Map.js [hunk 1]
```diff
@@ -53,13 +54,14 @@ const Map = ( props ) => {
...
   );
-};
+});
 
 Map.propTypes = {
   children: PropTypes.node,
   className: PropTypes.string,
   defaultBaseMap: PropTypes.string,
   mapEffect: PropTypes.func,
+  ref: PropTypes.shape({ current: any }),
```
Comment (colbyfayock) [inline src/components/Map.js:64]: can you update `any` to `PropTypes.any` and remove the import so it conforms with the rest of the proptype statements?

File: src/components/Map.js [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
 import { useConfigureLeaflet, useMapServices, useRefEffect } from 'hooks';
 import { isDomAvailable } from 'lib/util';
 
 const DEFAULT_MAP_SERVICE = 'OpenStreetMap';
 
-const Map = ( props ) => {
-  const { children, className, defaultBaseMap = DEFAULT_MAP_SERVICE, mapEffect, ...rest } = props;
-
-  const mapRef = useRef();
+const Map = React.forwardRef(( props ) => {
```
Comment (colbyfayock) [inline src/components/Map.js:10]: the ref argument actually needs to be included as a second argument next to props

https://reactjs.org/docs/forwarding-refs.html

if you look at the console when running the project, its throwing an error

image

**Comment by digvijaysrathore** [comment]
Thanks for the review. I fixed the issues you mentioned in the latest commit.

**Comment by colbyfayock** [comment]
looks good, thanks @digvijaysrathore 

@all-contributors please add @digvijaysrathore for code

**Pull request merged**